subsystem/infra_env_test.go

```diff
@@ -525,4 +527,15 @@ var _ = Describe("Infra_Env", func() {
 		_, err := userBMClient.Installer.UpdateInfraEnv(ctx, updateParams)
 		Expect(err).To(HaveOccurred())
 	})
+
+	It("fails when given invalid version", func() {
+		updateParams := &installer.UpdateInfraEnvParams{
+			InfraEnvID: infraEnvID,
+			InfraEnvUpdateParams: &models.InfraEnvUpdateParams{
+				OpenshiftVersion: swag.String("5.99"),
```

Comment:
Sorry, I might've missed this, but where is the version validated?

Reply:
The check already existed here https://github.com/openshift/assisted-service/blob/0fde766336a2e6854cdfb758583f15ec40380b25/internal/bminventory/inventory.go#L4991-L4993

Reply:
Gotcha, thanks for the pointer! 

That seems to check the infraenv's current cpu arch and openshift version, and not the new updated param's openshift version. Is that alright with your changes?

Reply:
Hm I missed that. But this test is passing ... Let me figure it out.

Reply:
Ah okay, it's failing in the ISO generation code here https://github.com/openshift/assisted-service/blob/0fde766336a2e6854cdfb758583f15ec40380b25/internal/bminventory/inventory.go#L5027-L5029 and later here https://github.com/openshift/assisted-service/blob/0fde766336a2e6854cdfb758583f15ec40380b25/internal/versions/osimages.go#L160

That's after we've already updated the database. I'll add a new validation call for this. Thanks for catching this @CrystalChun!

FWIW This is the message I get currently when I update the infraenv to a non-existent version:
```
    message: 'Failed to create image: No OS image for Openshift version 4.22 and architecture
      x86_64: The requested OS image for version (4.22) and CPU architecture (x86_64)
      isn''t specified in OS images list'
```

Reply:
Ah, so it fails down the line. That's good to know too!

Reply:
Added some new commits to validate this earlier.